packages/workflows/src/handlers/middlewares/create-products-prepare-create-prices-compensation.ts

```diff
@@ -0,0 +1,50 @@
+import { ProductTypes, WorkflowTypes } from "@medusajs/types"
+import { WorkflowArguments } from "../../helper"
+
+type ProductHandle = string
+type VariantIndexAndPrices = {
+  index: number
+  prices: WorkflowTypes.ProductWorkflow.CreateProductVariantPricesInputDTO[]
+}
+
+export async function createProductsPrepareCreatePricesCompensation({
+  data,
+}: WorkflowArguments<{
+  productsHandleVariantsIndexPricesMap: Map<
+    ProductHandle,
+    VariantIndexAndPrices[]
+  >
+  products: ProductTypes.ProductDTO[]
+}>) {
+  const productsHandleVariantsIndexPricesMap =
+    data.productsHandleVariantsIndexPricesMap
+  const products = data.products
+
+  const updatedProductsHandleVariantsIndexPricesMap = new Map()
+  productsHandleVariantsIndexPricesMap.forEach((items, productHandle) => {
+    const existingItems =
+      updatedProductsHandleVariantsIndexPricesMap.get(productHandle) ?? []
+
+    items.forEach(({ index }) => {
+      existingItems.push({
+        index,
+        prices: [],
+      })
+    })
+
+    updatedProductsHandleVariantsIndexPricesMap.set(productHandle, items)
+  })
+
+  return {
+    alias: "",
```

Comment:
question: Wouldn't be better for this to be optional instead of having to use an empty string?

Reply:
Yes i can do that, just wanted to not be based on an object with a value property as any object could have a value property and i wanted to prevent missmatch